Accidental breaking API change

With #873 we accidentally introduced a breaking API change. Before the change, you could pass a metric name filter as a lambda expression:

```java
registry.scrape(name -> name.equals("my_metric"));
```

Now, this fails:

```
reference to scrape is ambiguous
[ERROR]   both method scrape(io.prometheus.metrics.model.registry.PrometheusScrapeRequest) in io.prometheus.metrics.model.registry.PrometheusRegistry and method scrape(java.util.function.Predicate<java.lang.String>) in io.prometheus.metrics.model.registry.PrometheusRegistry match
```

Same for `(Multi)Collector.collect(name -> name.equals("..."))`.


@ganzuoni what do you think is a good way to fix this? I know your initial PR had a method `getRequestPath()`, which we removed because it wasn't needed. I guess we could add it again to the `PrometheusScrapeRequest` interface? What do you think?

--- a/prometheus-metrics-exporter-httpserver/src/main/java/io/prometheus/metrics/exporter/httpserver/HttpExchangeAdapter.java
+++ b/prometheus-metrics-exporter-httpserver/src/main/java/io/prometheus/metrics/exporter/httpserver/HttpExchangeAdapter.java
@@ -49,6 +49,17 @@ public class HttpExchangeAdapter implements PrometheusHttpExchange {
         public String getMethod() {
             return httpExchange.getRequestMethod();
         }
+
+        @Override
+        public String getRequestPath() {
+            URI requestURI = httpExchange.getRequestURI();
+            String uri = requestURI.toString();
+            int qx = uri.indexOf('?');
+            if (qx != -1) {
+                uri = uri.substring(0, qx);
+            }
+            return uri;
+        }
     }
 
     public class HttpResponse implements PrometheusHttpResponse {

--- a/prometheus-metrics-exporter-servlet-jakarta/src/main/java/io/prometheus/metrics/exporter/servlet/jakarta/HttpExchangeAdapter.java
+++ b/prometheus-metrics-exporter-servlet-jakarta/src/main/java/io/prometheus/metrics/exporter/servlet/jakarta/HttpExchangeAdapter.java
@@ -8,6 +8,7 @@ import jakarta.servlet.http.HttpServletResponse;
 
 import java.io.IOException;
 import java.io.OutputStream;
+import java.net.URI;
 import java.util.Enumeration;
 
 public class HttpExchangeAdapter implements PrometheusHttpExchange {
@@ -66,6 +67,24 @@ public class HttpExchangeAdapter implements PrometheusHttpExchange {
         @Override
         public String getMethod() {
             return request.getMethod();
+        }
+
+        @Override
+        public String getRequestPath() {
+            StringBuilder uri = new StringBuilder();
+            String contextPath = request.getContextPath();
+            if (contextPath.startsWith("/")) {
+                uri.append(contextPath);
+            }
+            String servletPath = request.getServletPath();
+            if (servletPath.startsWith("/")) {
+                uri.append(servletPath);
+            }
+            String pathInfo = request.getPathInfo();
+            if (pathInfo != null) {
+                uri.append(pathInfo);
+            }
+            return uri.toString();
         }
     }
 

--- a/prometheus-metrics-model/src/main/java/io/prometheus/metrics/model/registry/PrometheusScrapeRequest.java
+++ b/prometheus-metrics-model/src/main/java/io/prometheus/metrics/model/registry/PrometheusScrapeRequest.java
@@ -2,10 +2,16 @@ package io.prometheus.metrics.model.registry;
 
 /**
  * Infos extracted from the request received by the endpoint
- * 
  */
 public interface PrometheusScrapeRequest {
-	
-	String[] getParameterValues(String name);
 
+	/**
+	 * Absolute path of the HTTP request.
+	 */
+	String getRequestPath();
+
+	/**
+	 * See {@code jakarta.servlet.ServletRequest.getParameterValues(String name)}
+	 */
+	String[] getParameterValues(String name);
 }

